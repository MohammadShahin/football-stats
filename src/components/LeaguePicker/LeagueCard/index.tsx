import { type League } from '../../../types'
import styles from './LeagueCard.module.css'
import Image from 'next/image'

interface LeagueCardProps {
  league: League
  isActive: boolean
}

export default function LeagueCard({ league, isActive }: LeagueCardProps) {
  return (
    <div className={styles.card + (isActive ? ' ' + styles.active : '')}>
      <Image src={league.logo} alt={league.name} width={50} height={50} />
      <p>{league.name}</p>
      <div className={styles.flexRow}>
        <p>{league.country}</p>
        <Image src={league.flag} alt={league.country} width={30} height={30} />
      </div>
    </div>
  )
}
