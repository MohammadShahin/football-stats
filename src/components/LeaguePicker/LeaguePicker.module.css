.radioInput {
  position: absolute;
  opacity: 0;
  width: 0;
  height: 0;
}

.radioInput + * {
  cursor: pointer;
}

.radioInputSeason + * {
  background-color: #f1f1f1;
  width: 5em;
  height: 3rem;
  margin: 1rem;
  border-radius: 5px;
}

.grid {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-around;
    align-items: center;
    margin: 0 auto;
    text-align: center;    
}

.submitButton {
    background-color: #ff3e00;
    color: white;
    border: none;
    border-radius: 5px;
    padding: 1rem;
    margin: 1rem auto;
    margin-right: auto!;
    margin-left: auto!;
    font-size: 1.5rem;
    cursor: pointer;
}

.form {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin: 0 auto;
    text-align: center;
}
