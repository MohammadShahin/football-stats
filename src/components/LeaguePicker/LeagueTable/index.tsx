import { LeagueData } from '../../../types'
import styles from './LeagueTable.module.css'

interface LeagueTableProps {
  leagueData: LeagueData | null
}

export default function LeagueTable({ leagueData }: LeagueTableProps) {
  if (!leagueData) return null
  return (
    <table className={styles.leagueTable}>
      <thead>
        <tr>
          <th>Position</th>
          <th>Team</th>
          <th>Points</th>
        </tr>
      </thead>
      <tbody>
        {leagueData.standings.map((standing, index) => (
          <tr key={standing.team}>
            <td>{index + 1}</td>
            <td>{standing.team}</td>
            <td>{standing.points}</td>
          </tr>
        ))}
      </tbody>
    </table>
  )
}
