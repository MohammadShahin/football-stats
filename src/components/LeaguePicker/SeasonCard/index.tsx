import { type Season } from '../../../types'
import styles from './SeasonCard.module.css'

interface SeasonCardProps {
  season: Season
  isActive: boolean
}

export default function SeasonCard({ season, isActive }: SeasonCardProps) {
  return (
    <div className={styles.card + (isActive ? ' ' + styles.active : '')}>
      {season.name}
    </div>
  )
}
