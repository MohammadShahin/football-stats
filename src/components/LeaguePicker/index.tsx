import React from 'react';
import { ToastContainer, toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import LeagueCard from 'src/components/LeaguePicker/LeagueCard';
import { leagues, seasons } from '../../constants';
import styles from './LeaguePicker.module.css';
import LeagueTable from './LeagueTable';
import { League, LeagueData, Season } from '../../types';
import SeasonCard from './SeasonCard';
import { getStandings } from '../../utils/fetchStandings';

export default function LeaguePicker() {
  const [selectedLeague, setSelectedLeague] = React.useState<League | null>(
    null
  );
  const [selectedSeason, setSelectedSeason] = React.useState<Season | null>(
    null
  );
  const [leagueData, setLeagueData] = React.useState<LeagueData | null>(null);

  const [loading, setLoading] = React.useState(false);
  const [error, setError] = React.useState('');

  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();

    if (!selectedLeague) {
      toast('Please select a league', {
        position: 'top-right',
        hideProgressBar: true,
        closeOnClick: true,
        pauseOnHover: true,
        draggable: true,
        progress: undefined,
        theme: 'light',
        type: 'error',
      });
      return;
    }
    if (!selectedSeason) {
      toast('Please pick a season', {
        position: 'top-right',
        hideProgressBar: true,
        closeOnClick: true,
        pauseOnHover: true,
        draggable: true,
        progress: undefined,
        theme: 'light',
        type: 'error',
      });
      return;
    }

    setLoading(true);
    try {
      const newLeagueData: LeagueData = await getStandings(
        selectedLeague,
        selectedSeason
      );
      setLeagueData(newLeagueData);
    } catch {
      setError('Something went wrong');
    } finally {
      setLoading(false);
    }
  };

  return (
    <div>
      <ToastContainer
        position="top-right"
        autoClose={5000}
        hideProgressBar={true}
        newestOnTop={false}
        closeOnClick
        rtl={false}
        pauseOnFocusLoss
        draggable
        pauseOnHover
        theme="light"
      />
      <form className={styles.form} onSubmit={handleSubmit}>
        <p>Choose a league to see the standings</p>
        <div className="grid">
          {leagues.map((league) => (
            <label key={`league-${league.id}`}>
              <input
                className={`${styles.radioInput} ${styles.radioInputLeague}`}
                type="radio"
                checked={selectedLeague?.id === league.id}
                onChange={() => setSelectedLeague(league)}
              />
              <LeagueCard
                league={league}
                isActive={selectedLeague?.id === league.id}
              />
            </label>
          ))}
        </div>

        <br />
        <p>Choose the season you would like to discover</p>
        <div className="grid">
          {seasons.map((season) => (
            <label key={`season-${season.name}`}>
              <input
                className={`${styles.radioInput} ${styles.radioInputSeason}`}
                type="radio"
                checked={selectedSeason?.id === season.name}
                onChange={() => setSelectedSeason(season)}
              />
              <SeasonCard
                season={season}
                isActive={selectedSeason?.name === season.name}
              />
            </label>
          ))}
        </div>
        <button className={styles.submitButton} type="submit">
          Submit
        </button>
      </form>
      {loading && <p className={styles.loading}>Loading...</p>}
      {!!error && !loading && (
        <p className={styles.error}>Something went wrong</p>
      )}
      {!error && !loading && leagueData && (
        <LeagueTable leagueData={leagueData} />
      )}
    </div>
  );
}
