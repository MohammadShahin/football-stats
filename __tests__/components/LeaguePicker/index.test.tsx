import { render } from '@testing-library/react';
import LeaguePicker from 'src/components/LeaguePicker';

describe('LeaguePicker', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    jest.useFakeTimers();
  });

  it('renders correctly', () => {
    expect(() => render(<LeaguePicker />)).not.toThrow();
  });

  it('matches snapshot', () => {
    expect(render(<LeaguePicker />)).toMatchSnapshot();
  });
});
