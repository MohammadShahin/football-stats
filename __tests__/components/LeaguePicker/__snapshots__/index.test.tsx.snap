// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaguePicker matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="Toastify"
        />
        <form
          class="form"
        >
          <p>
            Choose a league to see the standings
          </p>
          <div
            class="grid"
          >
            <label>
              <input
                class="radioInput radioInputLeague"
                type="radio"
              />
              <div
                class="card"
              >
                <img
                  alt="Premier League"
                  data-nimg="1"
                  decoding="async"
                  height="50"
                  loading="lazy"
                  src="/_next/image?url=%2Fleagues%2Fpremier-league.png&w=128&q=75"
                  srcset="/_next/image?url=%2Fleagues%2Fpremier-league.png&w=64&q=75 1x, /_next/image?url=%2Fleagues%2Fpremier-league.png&w=128&q=75 2x"
                  style="color: transparent;"
                  width="50"
                />
                <p>
                  Premier League
                </p>
                <div
                  class="flexRow"
                >
                  <p>
                    England
                  </p>
                  <img
                    alt="England"
                    data-nimg="1"
                    decoding="async"
                    height="30"
                    loading="lazy"
                    src="/_next/image?url=%2Fflags%2Fengland.png&w=64&q=75"
                    srcset="/_next/image?url=%2Fflags%2Fengland.png&w=32&q=75 1x, /_next/image?url=%2Fflags%2Fengland.png&w=64&q=75 2x"
                    style="color: transparent;"
                    width="30"
                  />
                </div>
              </div>
            </label>
            <label>
              <input
                class="radioInput radioInputLeague"
                type="radio"
              />
              <div
                class="card"
              >
                <img
                  alt="La Liga"
                  data-nimg="1"
                  decoding="async"
                  height="50"
                  loading="lazy"
                  src="/_next/image?url=%2Fleagues%2Fla-liga.png&w=128&q=75"
                  srcset="/_next/image?url=%2Fleagues%2Fla-liga.png&w=64&q=75 1x, /_next/image?url=%2Fleagues%2Fla-liga.png&w=128&q=75 2x"
                  style="color: transparent;"
                  width="50"
                />
                <p>
                  La Liga
                </p>
                <div
                  class="flexRow"
                >
                  <p>
                    Spain
                  </p>
                  <img
                    alt="Spain"
                    data-nimg="1"
                    decoding="async"
                    height="30"
                    loading="lazy"
                    src="/_next/image?url=%2Fflags%2Fspain.png&w=64&q=75"
                    srcset="/_next/image?url=%2Fflags%2Fspain.png&w=32&q=75 1x, /_next/image?url=%2Fflags%2Fspain.png&w=64&q=75 2x"
                    style="color: transparent;"
                    width="30"
                  />
                </div>
              </div>
            </label>
            <label>
              <input
                class="radioInput radioInputLeague"
                type="radio"
              />
              <div
                class="card"
              >
                <img
                  alt="Bundesliga"
                  data-nimg="1"
                  decoding="async"
                  height="50"
                  loading="lazy"
                  src="/_next/image?url=%2Fleagues%2Fbundesliga.png&w=128&q=75"
                  srcset="/_next/image?url=%2Fleagues%2Fbundesliga.png&w=64&q=75 1x, /_next/image?url=%2Fleagues%2Fbundesliga.png&w=128&q=75 2x"
                  style="color: transparent;"
                  width="50"
                />
                <p>
                  Bundesliga
                </p>
                <div
                  class="flexRow"
                >
                  <p>
                    Germany
                  </p>
                  <img
                    alt="Germany"
                    data-nimg="1"
                    decoding="async"
                    height="30"
                    loading="lazy"
                    src="/_next/image?url=%2Fflags%2Fgermany.png&w=64&q=75"
                    srcset="/_next/image?url=%2Fflags%2Fgermany.png&w=32&q=75 1x, /_next/image?url=%2Fflags%2Fgermany.png&w=64&q=75 2x"
                    style="color: transparent;"
                    width="30"
                  />
                </div>
              </div>
            </label>
            <label>
              <input
                class="radioInput radioInputLeague"
                type="radio"
              />
              <div
                class="card"
              >
                <img
                  alt="Serie A"
                  data-nimg="1"
                  decoding="async"
                  height="50"
                  loading="lazy"
                  src="/_next/image?url=%2Fleagues%2Fserie-a.png&w=128&q=75"
                  srcset="/_next/image?url=%2Fleagues%2Fserie-a.png&w=64&q=75 1x, /_next/image?url=%2Fleagues%2Fserie-a.png&w=128&q=75 2x"
                  style="color: transparent;"
                  width="50"
                />
                <p>
                  Serie A
                </p>
                <div
                  class="flexRow"
                >
                  <p>
                    Italy
                  </p>
                  <img
                    alt="Italy"
                    data-nimg="1"
                    decoding="async"
                    height="30"
                    loading="lazy"
                    src="/_next/image?url=%2Fflags%2Fitaly.png&w=64&q=75"
                    srcset="/_next/image?url=%2Fflags%2Fitaly.png&w=32&q=75 1x, /_next/image?url=%2Fflags%2Fitaly.png&w=64&q=75 2x"
                    style="color: transparent;"
                    width="30"
                  />
                </div>
              </div>
            </label>
            <label>
              <input
                class="radioInput radioInputLeague"
                type="radio"
              />
              <div
                class="card"
              >
                <img
                  alt="Ligue 1"
                  data-nimg="1"
                  decoding="async"
                  height="50"
                  loading="lazy"
                  src="/_next/image?url=%2Fleagues%2Fligue-1.png&w=128&q=75"
                  srcset="/_next/image?url=%2Fleagues%2Fligue-1.png&w=64&q=75 1x, /_next/image?url=%2Fleagues%2Fligue-1.png&w=128&q=75 2x"
                  style="color: transparent;"
                  width="50"
                />
                <p>
                  Ligue 1
                </p>
                <div
                  class="flexRow"
                >
                  <p>
                    France
                  </p>
                  <img
                    alt="France"
                    data-nimg="1"
                    decoding="async"
                    height="30"
                    loading="lazy"
                    src="/_next/image?url=%2Fflags%2Ffrance.png&w=64&q=75"
                    srcset="/_next/image?url=%2Fflags%2Ffrance.png&w=32&q=75 1x, /_next/image?url=%2Fflags%2Ffrance.png&w=64&q=75 2x"
                    style="color: transparent;"
                    width="30"
                  />
                </div>
              </div>
            </label>
          </div>
          <br />
          <p>
            Choose the season you would like to discover
          </p>
          <div
            class="grid"
          >
            <label>
              <input
                class="radioInput radioInputSeason"
                type="radio"
              />
              <div
                class="card"
              >
                2019/2020
              </div>
            </label>
            <label>
              <input
                class="radioInput radioInputSeason"
                type="radio"
              />
              <div
                class="card"
              >
                2020/2021
              </div>
            </label>
            <label>
              <input
                class="radioInput radioInputSeason"
                type="radio"
              />
              <div
                class="card"
              >
                2021/2022
              </div>
            </label>
            <label>
              <input
                class="radioInput radioInputSeason"
                type="radio"
              />
              <div
                class="card"
              >
                2022/2023
              </div>
            </label>
          </div>
          <button
            class="submitButton"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="Toastify"
      />
      <form
        class="form"
      >
        <p>
          Choose a league to see the standings
        </p>
        <div
          class="grid"
        >
          <label>
            <input
              class="radioInput radioInputLeague"
              type="radio"
            />
            <div
              class="card"
            >
              <img
                alt="Premier League"
                data-nimg="1"
                decoding="async"
                height="50"
                loading="lazy"
                src="/_next/image?url=%2Fleagues%2Fpremier-league.png&w=128&q=75"
                srcset="/_next/image?url=%2Fleagues%2Fpremier-league.png&w=64&q=75 1x, /_next/image?url=%2Fleagues%2Fpremier-league.png&w=128&q=75 2x"
                style="color: transparent;"
                width="50"
              />
              <p>
                Premier League
              </p>
              <div
                class="flexRow"
              >
                <p>
                  England
                </p>
                <img
                  alt="England"
                  data-nimg="1"
                  decoding="async"
                  height="30"
                  loading="lazy"
                  src="/_next/image?url=%2Fflags%2Fengland.png&w=64&q=75"
                  srcset="/_next/image?url=%2Fflags%2Fengland.png&w=32&q=75 1x, /_next/image?url=%2Fflags%2Fengland.png&w=64&q=75 2x"
                  style="color: transparent;"
                  width="30"
                />
              </div>
            </div>
          </label>
          <label>
            <input
              class="radioInput radioInputLeague"
              type="radio"
            />
            <div
              class="card"
            >
              <img
                alt="La Liga"
                data-nimg="1"
                decoding="async"
                height="50"
                loading="lazy"
                src="/_next/image?url=%2Fleagues%2Fla-liga.png&w=128&q=75"
                srcset="/_next/image?url=%2Fleagues%2Fla-liga.png&w=64&q=75 1x, /_next/image?url=%2Fleagues%2Fla-liga.png&w=128&q=75 2x"
                style="color: transparent;"
                width="50"
              />
              <p>
                La Liga
              </p>
              <div
                class="flexRow"
              >
                <p>
                  Spain
                </p>
                <img
                  alt="Spain"
                  data-nimg="1"
                  decoding="async"
                  height="30"
                  loading="lazy"
                  src="/_next/image?url=%2Fflags%2Fspain.png&w=64&q=75"
                  srcset="/_next/image?url=%2Fflags%2Fspain.png&w=32&q=75 1x, /_next/image?url=%2Fflags%2Fspain.png&w=64&q=75 2x"
                  style="color: transparent;"
                  width="30"
                />
              </div>
            </div>
          </label>
          <label>
            <input
              class="radioInput radioInputLeague"
              type="radio"
            />
            <div
              class="card"
            >
              <img
                alt="Bundesliga"
                data-nimg="1"
                decoding="async"
                height="50"
                loading="lazy"
                src="/_next/image?url=%2Fleagues%2Fbundesliga.png&w=128&q=75"
                srcset="/_next/image?url=%2Fleagues%2Fbundesliga.png&w=64&q=75 1x, /_next/image?url=%2Fleagues%2Fbundesliga.png&w=128&q=75 2x"
                style="color: transparent;"
                width="50"
              />
              <p>
                Bundesliga
              </p>
              <div
                class="flexRow"
              >
                <p>
                  Germany
                </p>
                <img
                  alt="Germany"
                  data-nimg="1"
                  decoding="async"
                  height="30"
                  loading="lazy"
                  src="/_next/image?url=%2Fflags%2Fgermany.png&w=64&q=75"
                  srcset="/_next/image?url=%2Fflags%2Fgermany.png&w=32&q=75 1x, /_next/image?url=%2Fflags%2Fgermany.png&w=64&q=75 2x"
                  style="color: transparent;"
                  width="30"
                />
              </div>
            </div>
          </label>
          <label>
            <input
              class="radioInput radioInputLeague"
              type="radio"
            />
            <div
              class="card"
            >
              <img
                alt="Serie A"
                data-nimg="1"
                decoding="async"
                height="50"
                loading="lazy"
                src="/_next/image?url=%2Fleagues%2Fserie-a.png&w=128&q=75"
                srcset="/_next/image?url=%2Fleagues%2Fserie-a.png&w=64&q=75 1x, /_next/image?url=%2Fleagues%2Fserie-a.png&w=128&q=75 2x"
                style="color: transparent;"
                width="50"
              />
              <p>
                Serie A
              </p>
              <div
                class="flexRow"
              >
                <p>
                  Italy
                </p>
                <img
                  alt="Italy"
                  data-nimg="1"
                  decoding="async"
                  height="30"
                  loading="lazy"
                  src="/_next/image?url=%2Fflags%2Fitaly.png&w=64&q=75"
                  srcset="/_next/image?url=%2Fflags%2Fitaly.png&w=32&q=75 1x, /_next/image?url=%2Fflags%2Fitaly.png&w=64&q=75 2x"
                  style="color: transparent;"
                  width="30"
                />
              </div>
            </div>
          </label>
          <label>
            <input
              class="radioInput radioInputLeague"
              type="radio"
            />
            <div
              class="card"
            >
              <img
                alt="Ligue 1"
                data-nimg="1"
                decoding="async"
                height="50"
                loading="lazy"
                src="/_next/image?url=%2Fleagues%2Fligue-1.png&w=128&q=75"
                srcset="/_next/image?url=%2Fleagues%2Fligue-1.png&w=64&q=75 1x, /_next/image?url=%2Fleagues%2Fligue-1.png&w=128&q=75 2x"
                style="color: transparent;"
                width="50"
              />
              <p>
                Ligue 1
              </p>
              <div
                class="flexRow"
              >
                <p>
                  France
                </p>
                <img
                  alt="France"
                  data-nimg="1"
                  decoding="async"
                  height="30"
                  loading="lazy"
                  src="/_next/image?url=%2Fflags%2Ffrance.png&w=64&q=75"
                  srcset="/_next/image?url=%2Fflags%2Ffrance.png&w=32&q=75 1x, /_next/image?url=%2Fflags%2Ffrance.png&w=64&q=75 2x"
                  style="color: transparent;"
                  width="30"
                />
              </div>
            </div>
          </label>
        </div>
        <br />
        <p>
          Choose the season you would like to discover
        </p>
        <div
          class="grid"
        >
          <label>
            <input
              class="radioInput radioInputSeason"
              type="radio"
            />
            <div
              class="card"
            >
              2019/2020
            </div>
          </label>
          <label>
            <input
              class="radioInput radioInputSeason"
              type="radio"
            />
            <div
              class="card"
            >
              2020/2021
            </div>
          </label>
          <label>
            <input
              class="radioInput radioInputSeason"
              type="radio"
            />
            <div
              class="card"
            >
              2021/2022
            </div>
          </label>
          <label>
            <input
              class="radioInput radioInputSeason"
              type="radio"
            />
            <div
              class="card"
            >
              2022/2023
            </div>
          </label>
        </div>
        <button
          class="submitButton"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
